core/api.py

```diff
@@ -0,0 +1,117 @@
+import requests
+import json
+import re
+from core.util import load_tokens, refresh_token
+from core import constants
+
+def api_request(
+    url: str,
+    tokens: dict,
+    config_file = constants.CONFIG_FILE) -> dict:
+    """Makes an Fitbit API request.
+
+    To be called from within various API endpoint objects.
+    Will try to refresh the access token automatically if expired.
+
+    Args:
+        url (str): a premade API endpoint url
+        tokens (dict): containing access and refresh tokens
+        config_file (str, optional): file location of the config file. Defaults to 'config.toml'.
+
+    Raises:
+        Exception: any exception from API requesting.
+
+    Returns:
+        dict: from JSON data
+    """
+    HEADERS = {'Authorization': f"Bearer {tokens[constants.TOKEN_API_ACCESS_TOKEN_KEY]}",
+                 'accept':"application/json"}
+    
+    response = requests.get(url=url, headers=HEADERS)
+
+    # Refresh the tokens if access token expired
+    if response.status_code == 401:
+        error_type = json.loads(response.content)['errors'][0]['errorType']
+        if error_type == "expired_token":
+            # Get and save new tokens, the reload them
+            refresh_token(config_file)
+            tokens = load_tokens(config_file)
+            # Load new token into the header
+            HEADERS['Authorization'] = f"Bearer {tokens[constants.TOKEN_API_ACCESS_TOKEN_KEY]}"
+            response = requests.get(url=url, headers=HEADERS)
+                
+    # Fall through of the if statements prints and raises for other errors
+    try: 
+        response.raise_for_status()
+    except Exception as e:
+        print(f"Status Code: {response.status_code}")
+        print(f"Content: ")
+        print(f"{response.content}")
+        print(f"---")
+        raise e
+
+    return json.loads(response.content)
+
+def check_date_format_wrapper(date: str) -> str:
+    """Check dates are 'yyyy-MM-dd' or 'today'.
+
+    Args:
+        date (str): yyyy-MM-dd format (i.e. all digits) or 'today'

```

Comment:
The yyyy-MM-dd isn't actually all digits (since it includes dashes).  I think you could remove the "(i.e. all digits)" part since it's clear what the format should be already.

Reply:
done